pkg/storage/couchdb/couchdbstore.go

```diff
@@ -196,16 +197,111 @@ func (c *CouchDBStore) Get(k string) ([]byte, error) {
 		return nil, err
 	}
 
-	_, containsAttachment := destinationData["_attachments"]
+	return c.getStoredValueFromRawDoc(rawDoc, k)
+}
+
+// CreateIndex creates an index based on the provided CreateIndexRequest.
+// createIndexRequest.IndexStorageLocation refers to the design doc that the index should get placed in.
+// createIndexRequest.IndexName is the name for the index that will be stored in CouchDB.
+// createIndexRequest.WhatToIndex must be a valid index object as defined here:
+//   http://docs.couchdb.org/en/stable/api/database/find.html#db-index
+func (c *CouchDBStore) CreateIndex(createIndexRequest storage.CreateIndexRequest) error {
+	return c.db.CreateIndex(context.Background(), createIndexRequest.IndexStorageLocation,
+		createIndexRequest.IndexName, createIndexRequest.WhatToIndex)
+}
+
+// Query executes a query using the CouchDB _find endpoint.
+func (c *CouchDBStore) Query(findQuery string) (storage.ResultsIterator, error) {
+	resultRows, err := c.db.Find(context.Background(), findQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return &couchDBResultsIterator{store: c, resultRows: resultRows}, nil
+}
+
+type couchDBResultsIterator struct {
+	store      *CouchDBStore
+	resultRows *kivik.Rows
+	firstErr   error
+}
+
+// Next moves the pointer to the next value in the iterator. It returns false if the iterator is exhausted.
+// Note that the Kivik library automatically closes the kivik.Rows iterator if the iterator is exhaused.
+func (i *couchDBResultsIterator) Next() bool {
+	return i.resultRows.Next()
+}
+
+// Release releases associated resources. Release should always result in success
+// and can be called multiple times without causing an error.
+func (i *couchDBResultsIterator) Release() {
+	closeErr := i.resultRows.Close()
+
+	if i.firstErr == nil {
+		i.firstErr = closeErr
```

Comment:
Should just log the error.

Reply:
Similar comment as my other one - the Aries-Framework-Go Release method follows this "deferred" error pattern. Any concerns if I differ from this pattern in edge-core? (I don't have any... @fqutishat?)

Reply:
@DRK3  I think we can differ from this pattern in edge-core

Reply:
OK, thanks @fqutishat